<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1050</width>
    <height>700</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
    <item>
     <widget class="ScrollArea" name="scrollAreaMap">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>917</width>
         <height>611</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="MapWidget" name="mapWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1">
      <item>
       <widget class="ObjectEditWidget" name="objectEditor">
        <property name="title">
         <string>Object Editor</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupTileSet">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Tile Set</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QScrollArea" name="scrollAreaTileSet">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOn</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAsNeeded</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>83</width>
              <height>549</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="TileWidget" name="tileWidget" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1050</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
    <addaction name="separator"/>
    <addaction name="actionShowObjects"/>
    <addaction name="actionShowGrid"/>
    <addaction name="separator"/>
    <addaction name="actionHighlightWalkable"/>
    <addaction name="actionHighlightHoverable"/>
    <addaction name="actionHighlightMoveable"/>
    <addaction name="actionHighlightDestructible"/>
    <addaction name="actionHighlightShootThrough"/>
    <addaction name="actionHighlightPushOnto"/>
    <addaction name="actionHighlightSearchable"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionHowToUse"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionSelect"/>
    <addaction name="actionSelectArea"/>
    <addaction name="separator"/>
    <addaction name="actionDrawTiles"/>
    <addaction name="actionFloodFill"/>
    <addaction name="separator"/>
    <addaction name="actionPlacePlayer"/>
    <addaction name="actionPlaceRobot"/>
    <addaction name="actionPlaceKey"/>
    <addaction name="actionPlaceHiddenItem"/>
    <addaction name="actionPlaceTransporterPad"/>
    <addaction name="actionPlaceDoor"/>
    <addaction name="actionPlaceElevator"/>
    <addaction name="actionPlaceTrashCompactor"/>
    <addaction name="actionPlaceWaterRaft"/>
    <addaction name="actionDeleteObject"/>
    <addaction name="separator"/>
    <addaction name="actionValidateMap"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionSelectAll"/>
    <addaction name="actionCopyArea"/>
    <addaction name="actionCopyObjects"/>
    <addaction name="actionCutArea"/>
    <addaction name="actionCutObjects"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionFill"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Main Toolbar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionZoomIn"/>
   <addaction name="actionZoomOut"/>
   <addaction name="separator"/>
   <addaction name="actionSelect"/>
   <addaction name="actionSelectArea"/>
   <addaction name="actionDrawTiles"/>
   <addaction name="actionFloodFill"/>
   <addaction name="actionPlacePlayer"/>
   <addaction name="actionPlaceRobot"/>
   <addaction name="actionPlaceKey"/>
   <addaction name="actionPlaceHiddenItem"/>
   <addaction name="actionPlaceTransporterPad"/>
   <addaction name="actionPlaceDoor"/>
   <addaction name="actionPlaceElevator"/>
   <addaction name="actionPlaceTrashCompactor"/>
   <addaction name="actionPlaceWaterRaft"/>
   <addaction name="actionDeleteObject"/>
  </widget>
  <action name="actionHighlightWalkable">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Walkable</string>
   </property>
   <property name="toolTip">
    <string>Highlight walkable tiles</string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="actionHighlightHoverable">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Hoverable</string>
   </property>
   <property name="toolTip">
    <string>Highlight hoverable tiles</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="actionHighlightMoveable">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Movable</string>
   </property>
   <property name="toolTip">
    <string>Highlight movable tiles</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="actionHighlightDestructible">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Destructible</string>
   </property>
   <property name="toolTip">
    <string>Highlight destructible tiles</string>
   </property>
   <property name="shortcut">
    <string>4</string>
   </property>
  </action>
  <action name="actionHighlightShootThrough">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight Shoot &amp;Through</string>
   </property>
   <property name="toolTip">
    <string>Highlight tiles that can be shot through</string>
   </property>
   <property name="shortcut">
    <string>5</string>
   </property>
  </action>
  <action name="actionHighlightPushOnto">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Push Onto</string>
   </property>
   <property name="toolTip">
    <string>Highlight tiles onto which objects can be pushed</string>
   </property>
   <property name="shortcut">
    <string>6</string>
   </property>
  </action>
  <action name="actionHighlightSearchable">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Highlight &amp;Searchable</string>
   </property>
   <property name="toolTip">
    <string>Highlight searchable tiles</string>
   </property>
   <property name="shortcut">
    <string>7</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/zoom_in.svg</normaloff>:/zoom_in.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;In</string>
   </property>
   <property name="toolTip">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>+</string>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/zoom_out.svg</normaloff>:/zoom_out.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;Out</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
  </action>
  <action name="actionShowObjects">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Show Objects</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionClickEveryTile">
   <property name="text">
    <string>Click Every Tile</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/pointer.svg</normaloff>:/pointer.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Select</string>
   </property>
   <property name="toolTip">
    <string>Select objects on the map. Click on them to display and edit their properties.</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionDrawTiles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/paintbrush.svg</normaloff>:/paintbrush.svg</iconset>
   </property>
   <property name="text">
    <string>Draw &amp;Tiles</string>
   </property>
   <property name="toolTip">
    <string>Draw map tiles. Use this to draw the map background.</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
  <action name="actionShowGrid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Grid</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="actionLoadTilesetFromFile">
   <property name="text">
    <string>Load Tileset from &amp;File...</string>
   </property>
  </action>
  <action name="actionPlacePlayer">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/stickfigure.svg</normaloff>:/stickfigure.svg</iconset>
   </property>
   <property name="text">
    <string>Place &amp;Player</string>
   </property>
   <property name="toolTip">
    <string>Place player</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="actionPlaceRobot">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/hoverbot.svg</normaloff>:/hoverbot.svg</iconset>
   </property>
   <property name="text">
    <string>Place &amp;Robot</string>
   </property>
   <property name="toolTip">
    <string>Place robot</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionDeleteObject">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/cross.svg</normaloff>:/cross.svg</iconset>
   </property>
   <property name="text">
    <string>Delete &amp;Object</string>
   </property>
   <property name="shortcut">
    <string>X</string>
   </property>
  </action>
  <action name="actionPlaceDoor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place &amp;Door</string>
   </property>
   <property name="toolTip">
    <string>Place door</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionPlaceTransporterPad">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place Transporter P&amp;ad</string>
   </property>
   <property name="toolTip">
    <string>Place transporter pad</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionPlaceTrashCompactor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place Trash &amp;Compactor</string>
   </property>
   <property name="toolTip">
    <string>Place trash compactor</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionPlaceElevator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place &amp;Elevator</string>
   </property>
   <property name="toolTip">
    <string>Place elevator</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="actionPlaceWaterRaft">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place &amp;Water Raft</string>
   </property>
   <property name="toolTip">
    <string>Place water raft</string>
   </property>
   <property name="shortcut">
    <string>W</string>
   </property>
  </action>
  <action name="actionPlaceKey">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place &amp;Key</string>
   </property>
   <property name="toolTip">
    <string>Place key</string>
   </property>
   <property name="shortcut">
    <string>K</string>
   </property>
  </action>
  <action name="actionPlaceHiddenItem">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Place Hidden &amp;Item</string>
   </property>
   <property name="toolTip">
    <string>Place hidden item</string>
   </property>
   <property name="shortcut">
    <string>I</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open Map...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save Map</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save Map &amp;As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New Map</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionFloodFill">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/pour_bucket.svg</normaloff>:/pour_bucket.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Flood Fill with Tiles</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="actionSelectArea">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/dashed_rectangle.svg</normaloff>:/dashed_rectangle.svg</iconset>
   </property>
   <property name="text">
    <string>Select &amp;Area</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="actionCopyArea">
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy the area on the map selected with the select area tool.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionCutArea">
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
   <property name="toolTip">
    <string>Cut the area on the map selected with the select area tool</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionCopyObjects">
   <property name="text">
    <string>Copy &amp;Objects</string>
   </property>
  </action>
  <action name="actionCutObjects">
   <property name="text">
    <string>Cut O&amp;bjects</string>
   </property>
  </action>
  <action name="actionFill">
   <property name="text">
    <string>&amp;Fill</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="text">
    <string>Select &amp;All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionValidateMap">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/caution.svg</normaloff>:/caution.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Validate Map</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionRandomizeGrass">
   <property name="text">
    <string>Randomize Grass</string>
   </property>
   <property name="toolTip">
    <string>Randomizes light grass to look less uniform.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+G</string>
   </property>
  </action>
  <action name="actionRandomizeDirt">
   <property name="text">
    <string>Randomize Dirt</string>
   </property>
   <property name="toolTip">
    <string>Randomizes dirt to look less uniform</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
  <action name="actionLoadPetTileset">
   <property name="text">
    <string>Load Bundled PET Tileset</string>
   </property>
   <property name="toolTip">
    <string>Load the bundled tileset from the PET version of the game</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionHowToUse">
   <property name="text">
    <string>&amp;How To Use</string>
   </property>
   <property name="toolTip">
    <string>Show an explanation of how to use the map editor</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionLoadColorPetsciiTileset">
   <property name="text">
    <string>Load Bundled Color PETSCII Tileset</string>
   </property>
   <property name="toolTip">
    <string>Load the bundled tileset from the Color PESCII version of the game</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TileWidget</class>
   <extends>QWidget</extends>
   <header>tilewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MapWidget</class>
   <extends>QWidget</extends>
   <header>mapwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ObjectEditWidget</class>
   <extends>QGroupBox</extends>
   <header>objecteditwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ScrollArea</class>
   <extends>QScrollArea</extends>
   <header>scrollarea.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/res.qrc"/>
 </resources>
 <connections/>
</ui>
